Src/BootCamp.Chapter/Lesson4.cs

```diff
@@ -0,0 +1,148 @@
+            }
+        }
+
+        public static float GetBmi(float weight, float height)
+        {
+            if (height <= 0 || weight <= 0)
+            {
+                Console.WriteLine("Failed calculating BMI. Reason:");
+
+                if (height <= 0 && weight <= 0)
+                {
+                    Console.WriteLine($"Weight cannot be equal or less than zero, but was {weight}.{Environment.NewLine}Height cannot be less than zero, but was {height}.");
+                }
+                else if (height <= 0)
+                {
+                    Console.WriteLine($"Height cannot be equal or less than zero, but was {height}.");
+                }
+                else if (weight <= 0)
+                {
+                    Console.WriteLine($"Weight cannot be equal or less than zero, but was {weight}.");
+                }
+
+                return -1; 
+            }
+            
+
+
+            return weight / (float)Math.Pow(height, 2);
+        }
+
+        public static string GetString(string message)
+        {
+            Console.WriteLine(message);
+            var input = Console.ReadLine();
+            var validated = ValidateString(input); 
+            if (validated == "-")
+            {
+                Console.Write($"Name cannot be empty."); 
+            }
+
+            return validated;
+        }
+
+        public static string ValidateString(string input)
+        {
+            if (String.IsNullOrEmpty(input))
+            {
+                return "-";
+            }
+
+            return input; 
+             
+        }
+
+        public static int GetInt(string message)
+        {
+            Console.WriteLine(message);
+            var input = Console.ReadLine();
+            var validated = ValidateInt(input);
+            if (validated == -1)
+            {
+                DisplayErrorMessageForNumbers(input);
+            }
+
+            return validated;
+        }
+
+        public static void DisplayErrorMessageForNumbers(string input)
+        {
+            Console.Write($"\"{input}\" is not a valid number.");
+        }
+
+        private static int ValidateInt(string input)
+        {
+            if (String.IsNullOrEmpty(input))
+            {
+                return 0;
+            }
+
+            var isValid = int.TryParse(input, out int number);
+
+            if (isValid)
+            {
+                return number;
+            }
+
+            return -1;
+        }
+
+        public static float GetFloat(string message)
+        {
+            Console.WriteLine(message);
+            var input = Console.ReadLine();
+            var validated = ValidateFloat(input);
+            if (Math.Abs(validated - -1) <= 0.001)
+            {
+                DisplayErrorMessageForNumbers(input);
+            }
+
+            return validated;
+        }
+
+        private static float ValidateFloat(string input)
+        {
+            if (String.IsNullOrEmpty(input))
```

Comment:
My personal preference is to use string vs String when using methods. You could read more about this here for deeper knowledge https://stackoverflow.com/questions/7074/what-is-the-difference-between-string-and-string-in-c

Reply:
Thanks, I will pay more attention to this.